subsys/bluetooth/services/fast_pair/fp_gatt_service.c

```diff
@@ -98,11 +98,13 @@ struct msg_seekers_passkey {
 /* Reference to the Fast Pair information callback structure. */
 static const struct bt_fast_pair_info_cb *fast_pair_info_cb;
 
+#if CONFIG_BT_FAST_PAIR_CHRC_MODEL_ID
 static ssize_t read_model_id(struct bt_conn *conn,
 			     const struct bt_gatt_attr *attr,
 			     void *buf,
 			     uint16_t len,
 			     uint16_t offset);
+#endif /* CONFIG_BT_FAST_PAIR_CHRC_MODEL_ID */
```

Comment:
Is this necessary? We don't do it for `write_additional_data`. It should be opt-out anyway.

Reply:
Also, this change should belong to 1st commit not 2nd.

Reply:
Unfortunately, the additional data is covered by the following syntax:

```
	if (!IS_ENABLED(CONFIG_BT_FAST_PAIR_PN)) {
		ARG_UNUSED(write_additional_data);
		LOG_WRN("Action request not supported (Action request)");
		return -ENOTSUP;
	}
```

I added these ifdefs as the follow-up change due to compiler warning about the unused symbol

Reply:
Ok, then leave it as is.